src/tagstore.rs

```diff
@@ -0,0 +1,111 @@
+use serde::de::{Deserializer, Error};
+use serde::ser::{Serialize, Serializer};
+
+/// signals that a newtype is from a CBOR tag
+pub const CBOR_NEWTYPE_NAME: &str = "__cbor_tag";
+
+/// extensions for all serde serializers
+pub trait SerializerExt: Serializer {
+    /// basically serialize_newtype_struct with a cbor tag value
+    fn serialize_cbor_tagged<T: Serialize>(
+        self,
+        tag: u64,
+        value: &T,
+    ) -> core::result::Result<Self::Ok, Self::Error> {
+        tag_access::set_tag(Some(tag));
+        let r = self.serialize_newtype_struct(CBOR_NEWTYPE_NAME, value);
+        tag_access::set_tag(None);
+        r
+    }
+}
+
+impl<S: Serializer> SerializerExt for S {}
+
+/// extensions for all serde deserializers
+pub trait DeserializerExt<'de>: Deserializer<'de> {
+    /// expect the given cbor tag
+    fn expect_cbor_tag(&self, tag: u64) -> Result<(), Self::Error> {
+        match tag_access::get_tag() {
+            Some(t) if t == tag => Ok(()),
+            Some(_) => Err(Self::Error::custom("unexpected cbor tag")),
+            None => Err(Self::Error::custom("missing cbor tag!")),
+        }
+    }
+}
+
+impl<'de, D: Deserializer<'de>> DeserializerExt<'de> for D {}
+
+#[cfg(tags)]
+pub mod tag_access {
```

Comment:
So code above is always included but this is only included if the `tags` feature is enabled, right?
Shouldn't the code above also be disabled if tags are not used?

Reply:
I just wanted to get it to build somehow first. Will change.